source/common/buffer/buffer_impl.cc

```diff
@@ -130,8 +130,10 @@ void OwnedImpl::commit(RawSlice* iovecs, uint64_t num_iovecs) {
       slice_index--;
     }
     if (slice_index < 0) {
-      // There was no slice containing any data, so rewind the iterator at the first slice.
       slice_index = 0;
+      if (!slices_[0]) {
```

Comment:
why this change is needed?

Reply:
The existing code can't send a fully empty OwnedImpl because slices_[0] is nullptr.  This just patches that up.  Prior to this change completely empty Buffer::Instances were created differently (e.g. out of std::string) which did not have this problem.